### PR TITLE
Update search command in DG

### LINKED ISSUE #107: Update 'search' command in Developer Guide

### PR DESCRIPTION
**Pull request by kaiyichen**
Resolves #107

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -268,6 +268,55 @@ The following sequence diagram shows how the edit operation works:
 
 _{more aspects and alternatives to be added}_
 
+
+### Search applicant feature
+#### Rationale
```
Comment (zzhikai) [inline docs/DeveloperGuide.md:273]: Add empty line above for spacing

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -268,6 +268,55 @@ The following sequence diagram shows how the edit operation works:
...
+`Search`  allows for a quick view of applicant's information in the linkedout book.
+#### Implementation
```
Comment (zzhikai) [inline docs/DeveloperGuide.md:275]: Same, add an empty line above

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -268,6 +268,55 @@ The following sequence diagram shows how the edit operation works:
...
+
+Given below is an example usage scenario and how the search mechanism behaves at each step.
+
+Step 1. The user enters `search n/David` command to search for applicants in the linkedout book.
+
+Step 2. The input keywords will be passed into `SearchCommandParser` and creates a `NameContainsKeywordsPredicate` if the keywords are not empty.
+
+Step 3. The predicate is then passed into `Model#updateFilteredApplicantList` to filter and display applicants with partial name matching of "David" in the linkedout book.
+
+Step 4. The user enters `search j/Software Engineer` command to search for applicants in the linkedout book.
```
Comment (zzhikai) [inline docs/DeveloperGuide.md:291]:
```suggestion
Step 4. The user enters `search j/Software Engineer` command to search for applicants in the linkedout app.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -268,6 +268,55 @@ The following sequence diagram shows how the edit operation works:
...
+
+Step 5. The input keywords will be passed into `SearchCommandParser` and creates a `JobContainsKeywordsPredicate` if the keywords are not empty.
+
+Step 6. The predicate is then passed into `Model#updateFilteredApplicantList` to filter and display applicants with partial job name matching of "Software" or "Engineer"  in the linkedout book.
```
Comment (zzhikai) [inline docs/DeveloperGuide.md:295]:
```suggestion
Step 6. The predicate is then passed into `Model#updateFilteredApplicantList` to filter and display applicants with partial job name matching of "Software" or "Engineer"  in the linkedout app.
```

**Review by zzhikai — APPROVED** [review]
LGTM, other then some minor changes needed. Great work!

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -268,6 +268,55 @@ The following sequence diagram shows how the edit operation works:
 
 _{more aspects and alternatives to be added}_
 
+
+### Search applicant feature
+#### Rationale
+`Search`  allows for a quick view of applicant's information in the linkedout book.
+#### Implementation
+
+The proposed search mechanism is facilitated by `SearchCommandParser`. `SearchCommandParser` will map the creation of `KeywordsPredicate` based on the input prefix. `KeywordsPredicate` that supports the implementation:
```
Comment (RyanCheungJF) [inline docs/DeveloperGuide.md:277]:
```suggestion
The proposed search mechanism is facilitated by `SearchCommandParser`. `SearchCommandParser` will map the creation of `KeywordsPredicate` based on the input prefix. `KeywordsPredicate` supports the following implementations:
```

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -268,6 +268,55 @@ The following sequence diagram shows how the edit operation works:
 
 _{more aspects and alternatives to be added}_
 
+
+### Search applicant feature
+#### Rationale
+`Search`  allows for a quick view of applicant's information in the linkedout book.
```
Comment (RyanCheungJF) [inline docs/DeveloperGuide.md:274]:
```suggestion
`search` allows for a quick view of applicant's information in the application/ LinkedOUT.
```

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -268,6 +268,55 @@ The following sequence diagram shows how the edit operation works:
...
+#### Implementation
+
+The proposed search mechanism is facilitated by `SearchCommandParser`. `SearchCommandParser` will map the creation of `KeywordsPredicate` based on the input prefix. `KeywordsPredicate` that supports the implementation:
+* `NameContainsKeywordsPredicate` — Predicate which returns true if an applicant's full name matches partially with the input keyword.
+* `JobContainsKeywordsPredicate` — Predicate which returns true if an applicant's job name matches partially with the input keyword.
+
+These predicates assist the filtering of applicant list in the `Model` interface, specifically for  `Model#updateFilteredApplicantList` and `Model#getFilteredApplicantList()`.
```
Comment (RyanCheungJF) [inline docs/DeveloperGuide.md:281]:
```suggestion
These predicates assist the filtering of applicant list in the `Model` interface, specifically for  `Model#updateFilteredApplicantList()` and `Model#getFilteredApplicantList()`.
```

**Review by RyanCheungJF — COMMENTED** [review]
Just a few minor tweaks to phrasing, rest looks good!

**Pull request merged**